lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
What I don't understand, is why we must expose the internal workings of one specific platform (Ruby) to the end user. That whole thing seems broken to me. Perhaps I don't understand the use-case, but the way I see it, there's just no way you can create a Binary from a string unless a decoding is given. The "use the bytes of the string, whatever they are" is a misnomer, since some string implementations aren't even based on bytes. The conversion opens up a can of worms as it exposes the internal runtime implementation in a bad way (to my knowledge, 8-bit-ascii doesn't exist outside of Ruby).

Can you describe how this is supposed to be used?